libs/common/src/helpers/change-stream/validate-audit-details.test.ts

```diff
@@ -1,49 +1,149 @@
+        `userType must be 'tfm'`,
+      );
     });
 
-    it('throws if id property is a string but invalid ObjectId', () => {
-      expect(() => validateAuditDetails({ userType, id: '1234' })).toThrow(`Invalid ${userType} user id 1234`);
+    it('throws if the userType is system', () => {
+      expect(() => validateAuditDetailsAndUserType({ userType: 'system' }, 'tfm')).toThrow(`userType must be 'tfm'`);
     });
 
-    it('does not throw if id is a valid string', () => {
-      const returnedValue = validateAuditDetails({ userType, id: '1234567890abcdef12345678' });
+    it('does not throw if the userType is tfm and input is a valid string', () => {
+      const returnedValue = validateAuditDetailsAndUserType({ userType: 'tfm', id: '1234567890abcdef12345678' }, 'tfm');
 
       expect(returnedValue).toBe(undefined);
     });
 
-    it('does not throw if id is a valid ObjectId', () => {
-      const returnedValue = validateAuditDetails({ userType, id: new ObjectId('1234567890abcdef12345678') });
+    it('does not throw if the userType is tfm and input is a valid ObjectId', () => {
+      const returnedValue = validateAuditDetailsAndUserType({ userType: 'tfm', id: new ObjectId('1234567890abcdef12345678') }, 'tfm');
 
       expect(returnedValue).toBe(undefined);
     });
   });
 
-  describe('when userType is system', () => {
-    it('does not throw', () => {
-      const returnedValue = validateAuditDetails({ userType: 'system' });
+  describe('when validating userType is system', () => {
+    withValidateAuditDetailsTests((auditDetails) => validateAuditDetailsAndUserType(auditDetails, 'system'));
+
+    it('throws if the userType is tfm and id is a valid string', () => {
+      expect(() => validateAuditDetailsAndUserType({ userType: 'tfm', id: '1234567890abcdef12345678' }, 'system')).toThrow(`userType must be 'system'`);
+    });
+
+    it('throws if the userType is tfm and id is a valid ObjectId', () => {
+      expect(() => validateAuditDetailsAndUserType({ userType: 'tfm', id: new ObjectId('1234567890abcdef12345678') }, 'system')).toThrow(
+        `userType must be 'system'`,
+      );
+    });
+
+    it('throws if the userType is portal and id is a valid string', () => {
+      expect(() => validateAuditDetailsAndUserType({ userType: 'portal', id: '1234567890abcdef12345678' }, 'system')).toThrow(`userType must be 'system'`);
+    });
+
+    it('throws if the userType is portal and id is a valid ObjectId', () => {
+      expect(() => validateAuditDetailsAndUserType({ userType: 'portal', id: new ObjectId('1234567890abcdef12345678') }, 'system')).toThrow(
+        `userType must be 'system'`,
+      );
+    });
+
+    it('does not throw if the userType is system', () => {
+      const returnedValue = validateAuditDetailsAndUserType({ userType: 'system' }, 'system');
+
       expect(returnedValue).toBe(undefined);
     });
   });
 });
+
+function withValidateAuditDetailsTests(validationFunction: (auditDetails: unknown) => void) {
```

Comment:
ah this one covers the cases I mentioned earlier, ignore my previous comment :) nice tests